### PR TITLE
Add pre-commit step to beats pipelines

### PR DESCRIPTION
**Pull request by oakrizan**
## Proposed commit message
Added pre-commit check to beats pipelines to validate if no merge-conflicts exist.

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues

- Relates: https://github.com/elastic/ingest-dev/issues/3350

## Logs
Buildkite builds: https://buildkite.com/elastic/beats/builds?branch=oakrizan%3Aadd-pre-commit-plugin

### CODE REVIEW

File: .buildkite/auditbeat/auditbeat-pipeline.yml [hunk 1]
```diff
@@ -48,6 +48,17 @@ steps:
           - github_commit_status:
               context: "auditbeat: check/update"
 
+      - label: "Auditbeat: Run pre-commit"
+        command: "pre-commit run --all-files"
+        agents:
+          image: "docker.elastic.co/ci-agent-images/platform-ingest/buildkite-agent-beats-ci-with-hooks:latest"
+          cpu: "4000m"
```
Comment (dliappis) [inline .buildkite/auditbeat/auditbeat-pipeline.yml:55]: is 4 full cpu cores an overkill for running this test? Have you tested it with e.g. 2, is it enough?

Comment (alexsapran) [reply .buildkite/auditbeat/auditbeat-pipeline.yml:55]: What if we omit the cpu and mem limits?

**Review by dliappis — APPROVED** [review]
LGTM (assuming CI is green).

In a future PR we could add the pre-commit step also to x-pack/agentbeat: https://github.com/elastic/beats/blob/main/.buildkite/x-pack/pipeline.xpack.agentbeat.yml#L9

**Pull request merged**